OFF-EP 0000

SimonBoothroyd opened the pull request:
## Description

This PR tracks the initial draft of this repositories contents. I opted to include the main documents in this PR rather than the main initial commit so as to avoid putting very WIP standards and procedures front and center from the start. 

See https://github.com/openforcefield/openff-toolkit/issues/741 for more details for the rationale behind this repo + PR.

cc @j-wags @mattwthompson @Yoshanuikabundi 

## Status

- [x] Ready to go

mattwthompson commented:
I've fleshed out a few more sections and tinkered with some details (in ways that may be personal preference and not consensus).

A major outstanding question from my perspective is how to constitute technical committees
* One overarching OpenFF committee OR one for each relevant domain (SMIRNOFF, molecule representations, etc.) OR a mixture of the two? It's tempting to think about elaborate administrative structures but the most efficient would probably be just to have one that covers the entire organization, akin to a steering committee.
* Should these committees be limited to "internal" people, i.e. OpenFF PIs, engineers, and scientists? I would think it's the responsibility of direct stakeholders to also consider external stakeholders, and therefore we would not need to **directly** involve many people in the process, but appropriately consider their needs while still holding the final approval power.
* What is the role of advisory boards in these decisions? I don't have a good sense of this but my guess is the answer to this question is similar to the answer above.

j-wags commented:
> One overarching OpenFF committee OR one for each relevant domain

Let's initially just assign the OpenFF governing board as the uber-committee, and let them decide the number/content of the subcommittees.

j-wags commented on docs/enhancement-proposals/off-ep-0000.md:
```diff
@@ -0,0 +1,92 @@
+
+All OFF-EPs will be resolved as either *Rejected*, *Accepted* or *Deferred*
+
+An OFF-EP is accepted upon consensus approval of the stakeholders.
+The authors of a NEP are responsible for defining a consensus threshold.
+
+[comment]: <> (Potentially tricky; PEP and NEP do not make acceptance guidelines clear)
+
+In some cases unanimity may be necessary, in others some strong majority may be sufficient.
+A stakeholder(s) holding genuine reservations about any portion of the EP may block its acceptance, overriding the acceptance criteria laid out by the author(s).
+
+In unusual cases, OpenFF PIs may be asked to decide whether a controversial NEP is `Accepted` and may elect to resolve the status to `Accepted` or `Deferred`.
```
This is complicated. Let's keep it simple for now.

```suggestion
An OFF-EP is accepted upon unanimous approval of @SimonBoothroyd @j-wags @davidlmobley and @jchodera. 
```

davidlmobley replied:
This seems reasonable, though what if we couldn't agree? That scenario may not need formalization though.

davidlmobley replied:
Perhaps add "This acceptance criterion is expected to be updated/amended as the community grows and changes."

j-wags commented on docs/enhancement-proposals/off-ep-template.md:
```diff
@@ -0,0 +1,70 @@
+
+**Stakeholders:** &lt;list of stakeholders that would be affected by this proposal>
+
+**Acceptance criteria:** <Unanimity | Majority>
+
+[comment]: <> (What other acceptance criteria should be templated? Is this a good idea? See "How" in EP 0000)
```
Again, probably best to keep it simple early on.

```suggestion
**Acceptance criteria:** <Unanimity>

[comment]: <> (What other acceptance criteria should be templated? Is this a good idea? See "How" in EP 0000)
```

j-wags reviewed (approved):
I thought about this for a while and typed up a lot, but I think this is all too abstract until we start actually iterating. Because everyone's so busy, one thing I would change is to make it unambiguous who is required for a vote, at least until we reach our stride and can form subcommittees.

davidlmobley commented on docs/enhancement-proposals/off-ep-0000.md:
```diff
@@ -0,0 +1,92 @@
+1. **Informational** OFF-EPs provide general, non-binding guidance to the community but do not propose a new feature.
+1. **Process** OFF-EPs describe a change that may not be technical in nature but requires community input.
+
+[comment]: <> (The types are copied from PEPs, which are more elaborate. Can we just drop this section? i.e. is it worth the overhead to separate out EPs by these distinctions?)
```
It seems like it might be helpful to have this distinction.

davidlmobley commented on docs/enhancement-proposals/off-ep-0000.md:
```diff
@@ -0,0 +1,92 @@
+All OFF-EPs will be resolved as either *Rejected*, *Accepted* or *Deferred*
+
+An OFF-EP is accepted upon consensus approval of the stakeholders.
+The authors of a NEP are responsible for defining a consensus threshold.
```
What's a "NEP"?

mattwthompson replied:
Just a copy-paste (NumPy EP) that I neglected to fix 🙂

davidlmobley commented on docs/enhancement-proposals/off-ep-0000.md:
```diff
@@ -0,0 +1,92 @@
+An OFF-EP is accepted upon consensus approval of the stakeholders.
+The authors of a NEP are responsible for defining a consensus threshold.
+
+[comment]: <> (Potentially tricky; PEP and NEP do not make acceptance guidelines clear)
```
That is indeed tricky. No real suggestions if it's a community thing; maybe we can give some guidance about how long the feedback period needs to be or something?

davidlmobley commented on docs/enhancement-proposals/off-ep-0000.md:
```diff
@@ -0,0 +1,92 @@
+
+This document is based based upon the [`NEP 0`](https://github.com/numpy/numpy/blob/master/doc/neps/nep-0000.rst),  [`CFEP 01`](https://github.com/conda-forge/cfep/blob/master/cfep-01.md), and [`PEP 1`](https://www.python.org/dev/peps/pep-0001/)
+
+This document is explicitly explicitly [CC0 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
```
repeated "explicitly"

davidlmobley reviewed (changes requested):
In general looks good to me; I'm supportive pending minor edits.

mattwthompson commented:
@davidlmobley - I've come around @j-wags's suggestion of starting with simple criteria, but doing so with the understanding that they may change once we try them out in practice. If you were confused about murky criteria for accepting EPs, you're not alone - the lack of concretely documented processes for  PEPs becoming accepted seem to be intentional, and there's sometimes confusion from the community about who has the "final say" in any. (I'm not sure about the analogs for NumPy and conda-forge EPs.) So I think it's fair to not set our goals too lofty here but instead be flexible for future changes.

**I think the path forward here to get approvalfrom the 4 current SMIRNOFF committee members** (@j-wags, @davidlmobley , @SimonBoothroyd , @jchodera from https://github.com/openforcefield/openff-toolkit/issues/741#issuecomment-783601422) **that this document is a workable starting point** and this PR should be merged. From there, we can clarify as needed some of the remaining details (the roles of the governing board and other other committees, processes for OFF-EPs being approved, etc.) as changes to this EP or new EPs. This would also let us act on updates to the SMIRNOFF spec, which was discussed several months ago as the rationale for all of this. This is somewhat timely as we want clearly-documented spec updates to help with some corner cases that have been plaguing the toolkit lately (https://github.com/openforcefield/openff-toolkit/issues/716, https://github.com/openforcefield/openff-toolkit/issues/882, https://github.com/openforcefield/openff-toolkit/issues/734, probably more coming down the pipeline with other infrastructure work and some users doing experiments with non-bonded functional forms) and better ways of incorporating feedback in general.

mattwthompson commented:
Tracking approval here:
- [ ] @davidlmobley 
- [ ] @jchodera 
- [x] @j-wags 
- [ ] @SimonBoothroyd

SimonBoothroyd commented:
Because OFF-EPs may also track procedural or community based changes I've added @karmencj to the approvers list and modified the acceptance criteria to a simple major. The main rationale is that not all of the approvers will need to be involved in all OFF-EPs (e.g. I'm sure @davidlmobley and @karmencj will not be particularly interested in certain software specific OFF-EPs.)
